sympy/geometry/ellipse.py

```diff
@@ -656,7 +659,7 @@ def intersection(self, o):
         elif isinstance(o, LinearEntity3D):
                 raise TypeError('Entity must be two dimensional, not three dimensional')
         else:
-            raise TypeError('Wrong type of argument were put')
+            raise TypeError('Intersection not handled for %s' % func_name(o))
```

Comment:
This statement is much better than previous. We need to unified this in whole `geometry` package , like #12475